src/components/DevelopmentGuide/V4MigrationGuide.mdx

```diff
@@ -0,0 +1,256 @@
+    | Desktop | 1280px |
+  </Box>
+
+ <Box width="100%">
+    <Heading level="h5" mb={4} size="heading6" textAlign="center">New Breakpoints</Heading>
+    {/* prettier-ignore */}
+    | Name | Value |
+    | :--- | :---- |
+    | Small mobile (base) | 0 |
+    | Large mobile | 480px |
+    | Small tablet | 768px |
+    | Large tablet | 1024px |
+    | Desktop | 1280px |
+  </Box>
+</Flex>
+
+#### Rename `tableTextSize` to `fontSize`.
+
+We updated the `tableTextSize` prop in `Table` to `fontSize` to align more closely with the CSS property.
+
+## Non-Breaking Changes
+
+These updates enhance flexibility, stability, and developer experience without requiring changes to your existing code.
+
+#### A `useResponsiveSpacing` hook was introduced.
+
+[`useResponsiveSpacing`](../?path=/docs/hooks-useresponsivespacing--docs) exports spacing values to generate consistent layouts.
+We recommend its usage for custom layout and components.
+
+#### The `id` prop is now optional for all components.
+
+For components that require `id`s to meet accessibility requirements, one will automatically be generated using React's `useId`
+hook if none is passed.
+
+#### All components now include a default `data-testid` attribute on their top-level element.
+
+The `data-testid` value follows the pattern `ds-[componentName]` (e.g., `ds-accordion`, `ds-button`).
+
+#### Component prop interfaces have been significantly expanded to accept a wider range of values.
+
+This includes standard native HTML attributes (e.g., `aria-label`, `type` for inputs, `tabIndex`) and all Chakra UI style props
+(e.g., `p`, `m`, `bg`, `color`, `sx`).
+
+Refer to the dedicated note above each component's props table for details on how its types were extended.
+
+#### We're no longer throwing errors.
+
+Components like `Breadcrumbs`, `Heading`, and `Image` no longer throw errors (crashing your application) when missing required prop
+values.
+
+#### There are no warnings for invalid children.
+
+Components that expect specific child types (e.g., `RadioGroup` expecting `Radio` children) will no longer log warnings reducing
+console noise during development.
+
+#### We removed unused code.
+
+The `Header` component was needed for a specific problem that no longer exists and has been removed in v4.
+
+`FilterBar` was deprecated in favor of `FilterBarInline` and `FilterBarPopup`.
+
+The `Autosuggest` documentation was originally included as a guideline for using `react-autosuggest`, but is no longer recommended
+due to technical complexity and UX consistency concerns and has been removed in v4.
+
+#### Other optimizations
+
+- We've updated our Storybook version, and given the doc headers a new look and feel.
+- We've reduced `React.Children.map` usage in hopes of a performance gain in the components that were using it.
+- We've updated some of our internal naming for improved consistency.
+- We're utilizing object destructuring for arguments, enhancing readability and maintainability.
```

Comment:
Perhaps say "_internal_ readability and ..." since this just affects our codebase.